app/controllers/groupings_controller.rb

```diff
@@ -34,14 +51,27 @@ def update
     redirect_to edit_grouping_path
   end
 
+  def destroy
+    if @grouping.destroy
+      redirect_to groupings_path
+    else
+      flash[:error] = @grouping.errors.full_messages
+      redirect_to edit_grouping_path
+    end
+  end
+
   private
 
   def load_grouping
     @grouping = Grouping.find(params.require(:id))
   end
 
+  def new_grouping_params
+    params.require(:grouping).permit(:parent_grouping_id, :name)
```

Comment:
Any reason why you're not using `grouping_params` instead?

Reply:
Because `grouping_params` permits a background but they should be able to upload a background when creating a new grouping. Maybe it's unnecessary?